include/RAJA/util/Layout.hpp

```diff
@@ -232,9 +235,9 @@ struct LayoutBase_impl<camp::idx_seq<RangeInts...>, IdxLin, StrideOneDim> {
                                               Indices &&... indices) const
   {
 #if defined(RAJA_BOUNDS_CHECK_INTERNAL)
-    RAJA::Index_type totSize{1};
+    IdxLin totSize{1};
     for(size_t i=0; i<n_dims; ++i) {totSize *= sizes[i];};
-    if(linear_index < 0 || linear_index >= totSize) {
+    if(totSize > 0 && (linear_index < 0 || linear_index >= totSize)) {
       printf("Error! Linear index %ld is not within bounds [0, %ld]. \n",
              static_cast<long int>(linear_index), static_cast<long int>(totSize-1));
       RAJA_ASSERT(linear_index < 0 || linear_index >= totSize);
```

Comment:
Should we add the totSize > 0 condition to the assert?

Reply:
Actually, I'm wondering if we should change the name of RAJA_ASSERT to RAJA_HALT_KERNEL. Only when assert is defined, does it do an assert otherwise it just exits the kernel. This was an issue with the NDEBUG definition by CMAKE. If NDEBUG is defined, then assert is a no-op.

Reply:
Is RAJA_EXIT_KERNEL better??

Reply:
Yes, I like that better. I can change that here.